todoist.el

```diff
@@ -278,6 +279,18 @@ CACHE is optional param to get projects from cache."
          (name (completing-read "Select project: " (mapcar 'todoist--project-name projects))))
     (-find (lambda (x) (equal (todoist--project-name x) name)) projects)))
 
+(defun todoist--refile-handler ()
+  "Delete todoist task when refiled elsewhere. Called via `org-after-refile-insert-hook'.
+
+CAVEAT: Does not correctly handle refiling a region with multiple tasks. In this case, only the
+first task will be deleted."
+  (when (and (todoist--under-cursor-task-id)
+             (not (equal todoist-buffer-name (buffer-name))) ;; Refiled to another buffer.
+             (not org-refile-keep))
+    ;; Using query directly instead of `todoist-delete-task' for performance. Since the task was
+    ;; refiled there's no need to refresh the todoist buffer to remove it.
+    (todoist--query "DELETE" (format "/tasks/%s" (todoist--under-cursor-task-id)))))
```

Comment:
Does this seem reasonable to you? Calling `toodist--query` directly instead of `todoist-delete` saves close to 5 seconds for me, and that's with only a couple hundred tasks, a 1 gb/s internet connection, and a fast computer. That 5 seconds adds up quickly when refiling a bunch of tasks one by one.

Reply:
Yeah that's totally reasonable. `todoist-delete` is meant to be an interactive function and it's clearly not the intent here. It's a single line duplicated and it's easy to track, I say perfectly fine.